src/main/java/seedu/task/model/tracker/TaskByTagTracker.java

```diff
@@ -0,0 +1,71 @@
+package seedu.task.model.tracker;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import seedu.task.model.Model;
+import seedu.task.model.tag.Tag;
+import seedu.task.model.task.Task;
+
+/**
+ * Implementation of the stats feature to sort by occurrences of each tag.
+ */
+public class TaskByTagTracker {
+
+    /**
+     * Outputs in descending order the most frequent tag and its corresponding number of occurrence.
+     * @param model interface to obtain data from
+     * @return output to be printed
+     */
+    public static String sort(Model model) {
+        List<Task> lastShownList = model.getFilteredTaskList();
+        int size = lastShownList.size();
+        HashMap<Tag, Integer> counter = new HashMap<>();
+
+        for (int i = 0; i < size; i++) {
+            Set<Tag> tagSet = lastShownList.get(i).getTags();
+            Iterator<Tag> itr = tagSet.iterator();
+            while (itr.hasNext()) {
+                counter = editHashMap(counter, itr.next());
+            }
+        }
+        PriorityQueue<TagComparator> ordering = createOrdering(counter);
+        String output = "";
+        int idx = 0;
+        int tags = ordering.size();
+
+        while (idx < 10 && tags > 0) {
+            TagComparator tc = ordering.poll();
+            output = appendToBack(output, idx, tc);
+            tags--;
+            idx++;
```

Comment:
Line 25 - 35 seem to be quite decoupled from Line 36 - 45. I think there is some potential for abstraction here. Might I suggest that we abstract them into separate methods so that this method doesn't seem too long? My suggestion would be to abstract line 36 to 46 into a separate method and just return it.

Reply:
Good point! I've made the changes :)